src/components/App/App.js

```diff
@@ -19,10 +24,7 @@ class App extends React.Component {
 					</div>
 					<button
 						className="button button-snackchat"
-						onClick={() => {
-							this.props.setSendWithPhoto(true);
-							this.props.history.push('/disclaimer');
-						}}>
+						onClick={() => scanItem(true)}>
```

Comment:
To piggyback onto this...

We try and avoid creating functions in render since it can be called a lot and this can become a performance bottleneck in larger applications. It's a good habit to define these once (e.g. class method) and pass a reference to the function.

I think Emma has a good point of extracting to `handleSnackChatButtonClick` not only because of using a reference but also that it's actually what the user is doing. `scanItem(true | false)` is just an implementation detail of how snack chat is meant to work for now, if the requirements were changed in the future this would get messy quite quickly.

So it'd end up looking something like...

```
handleSnackChatClick() {
}

handleReminderClick() {
}

render() {
  onClick={this.handleSnackChatClick}
  onClick={this.handleReminderClick}
}
```

Ideally we'd move away from `scanItem(true | false)` to something like `sendSnackChat` and `sendReminder` that have different implementations. I think we have to be careful with keeping things "DRY", a few lines saved doesn't make up for the wrong abstraction